mslib/mscolab/server.py

```diff
@@ -642,6 +643,56 @@ def import_permissions():
                     "message": message})
 
 
+@APP.route('/reset_password/<token>', methods=['GET', 'POST'])
+def reset_password(token):
+    try:
+        email = confirm_token(token, expiration=86400)
+    except TypeError:
+        return jsonify({"success": False}), 401
+    if email is False:
+        return jsonify({"success": False}), 401
+    user = User.query.filter_by(emailid=email).first_or_404()
+    if user.confirmed:
+        form = ResetPasswordForm()
+        if form.validate_on_submit():
+            try:
+                user.hash_password(form.confirm_password.data)
+                db.session.commit()
+                flash('Password reset Success. Please login by the user interface.', 'category_success')
+                return render_template('user/status.html')
+            except IOError:
+                flash('Password reset failed. Please try again later', 'category_danger')
+        return render_template('user/reset_password.html', form=form)
+    else:
+        return jsonify({"success": False}), 401
+
+
+@APP.route("/reset_request", methods=['GET', 'POST'])
+def reset_request():
+    form = ResetRequestForm()
+    if form.validate_on_submit():
+        # Check wheather user exists or not based on the db
+        user = User.query.filter_by(emailid=form.email.data).first()
+        if user:
+            try:
+                username = user.username
+                token = generate_confirmation_token(form.email.data)
+                reset_password_url = url_for('reset_password', token=token, _external=True)
+                html = render_template('user/reset_confirmation.html',
+                                       reset_password_url=reset_password_url, username=username)
+                subject = "Password reset request"
+                send_email(form.email.data, subject, html)
+                flash('''Your password reset request has been sent successfully.
+                Please check your email for further instructions.''', 'category_success')
+                return render_template('user/status.html')
+            except IOError:
+                flash('''We apologize, but it seems that there was an issue sending
+                 your request email. Please try again later.''', 'category_info')
+        else:
+            flash('Sorry. We could not find your account', 'category_danger')
```

Comment:
We should always protect a lookup who has an account on the server. Unfortunately anyone can enter values.
This means we should give the same message as when we successful requested a password reset. 

The message should be something like: "An email was sent if this user account exists."